clifford/_layout.py

```diff
@@ -574,7 +574,7 @@ def _gen_complement_func(self, omt):
 
         @_numba_utils.njit
         def comp_func(Xval):
-            Yval = np.zeros(dims)
+            Yval = np.zeros(dims, dtype=Xval.dtype)
```

Comment:
Can you add a test to `test_clifford::TestClifford.test_unary_op_preserves_dtype` for this?